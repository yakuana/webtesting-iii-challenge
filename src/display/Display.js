import React from 'react';

const Display = ({ closed, locked }) => {
  const closedClass = `led ${closed ? 'red-led' : 'green-led'}`;
  const lockedClass = `led ${locked ? 'red-led' : 'green-led'}`;

  return (
    <div className="display panel">
      <div data-testid="display-un-lock" className={lockedClass}>{locked ? 'Locked' : 'Unlocked'}</div>
      <div data-testid="display-opn-close" className={closedClass}>{closed ? 'Closed' : 'Open'}</div>
    </div>
  );
};

Display.defaultProps = {
  closed: false,
  locked: false,
};

export default Display;
