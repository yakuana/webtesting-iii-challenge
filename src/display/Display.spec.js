// Test away!

// react imports 
import React from "react";
import renderer from "react-test-renderer"; 
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";  

// components 
import Display from "./Display.js"; 

describe("<Display />", () => {

    it("shows the controls and display", () => { 
        // generates a DOM tree
        const displayTree = renderer.create(<Display />); 

        // snapshots are a JSON representation of the DOM tree
        expect(displayTree.toJSON()).toMatchSnapshot();
    }); 

    it("displays 'Closed' if the `closed` prop is `true`", () => {
        const { getByTestId } = render(<Display closed={true} />)

        expect(getByTestId("display-opn-close")).toHaveTextContent(/Closed/i)
    });

    it("displays 'Open' if `closed` prop is false", () => {
        const { getByTestId } = render(<Display closed={false} />)

        expect(getByTestId("display-opn-close")).toHaveTextContent(/Open/i)
    });

    it("displays 'Locked' if the `locked` prop is `true`", () => {
        const { getByTestId } = render(<Display locked={true} />)

        expect(getByTestId("display-un-lock")).toHaveTextContent(/Locked/i)
    });

    it("displays 'Unlocked' if `locked` prop is false", () => {
        const { getByTestId } = render(<Display locked={false} />)

        expect(getByTestId("display-un-lock")).toHaveTextContent(/Unlocked/i)
    });

    it("when `closed` use the `red-led` class", () => {
        const { getByTestId } = render(<Display closed={true} />)

        expect(getByTestId("display-opn-close")).toHaveClass("red-led")
    });

    it("when `locked` use the `red-led` class", () => {
        const { getByTestId } = render(<Display locked={true} />)

        expect(getByTestId("display-un-lock")).toHaveClass("red-led")
    });

    it("when `open` use the `green-led` class", () => {
        const { getByTestId } = render(<Display closed={false} />)

        expect(getByTestId("display-opn-close")).toHaveClass("green-led")
    });

    it("when `unlocked` use the `green-led` class", () => {
        const { getByTestId } = render(<Display locked={false} />)

        expect(getByTestId("display-un-lock")).toHaveClass("green-led")
    });


})