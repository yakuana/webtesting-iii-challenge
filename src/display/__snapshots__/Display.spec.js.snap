// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> shows the controls and display 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="display-un-lock"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="display-opn-close"
  >
    Open
  </div>
</div>
`;
